modules/adfs/lib/IdP/ADFS.php

```diff
@@ -38,6 +38,7 @@ public static function receiveAuthnRequest(SimpleSAML_IdP $idp) {
 			'ForceAuthn' => $forceAuthn,
 			'isPassive' => $isPassive,
 			'adfs:wctx' => $wctx,
+			'adfs:wreply' => (array_key_exists('wreply',$_GET)) ? $_GET['wreply'] : ''
```

Comment:
The redirection only occurs when successful login, so not a really big issue, but still it must be addressed..

I will look into the whitelisting of hosts... Additionally i think i add a parameter in metadata which must be explicit set to allow the use ud adfs:wreply